src/wallet/rpc/addresses.cpp

```diff
@@ -135,13 +200,31 @@ RPCHelpMan setlabel()
 
     LOCK(pwallet->cs_wallet);
 
-    CTxDestination dest = DecodeDestination(request.params[0].get_str());
+    auto address = request.params[0].get_str();
+
+    auto decoded = DecodeDestinationIndicatingSP(address);
+    CTxDestination dest{std::get<0>(decoded)};
+    bool silent_payment{std::get<1>(decoded)};
+    int32_t silent_payment_index{std::get<2>(decoded)};
```

Comment:
a0c5c6cf29835ee8315356e050c01db9252e9dc1
use structured binding

Reply:
Done in https://github.com/bitcoin/bitcoin/pull/24897/commits/7a03d0cd787ac53d44206744c0f85a22399f71e3.